lib/cucumber/core/test/result.rb

```diff
@@ -177,6 +177,24 @@ def ok?(be_strict = false)
           end
         end
 
+        class Ambiguous < Raisable
+          include Result.query_methods :ambiguous
+
+          def describe_to(visitor, *args)
+            visitor.failed(*args)
+            visitor.duration(duration, *args)
+            self
+          end
+
+          def to_s
+            "F"
```

Comment:
> starting with the even more basic case of failure

In that case, an `Ambiguous` class should not be defined here, so defining it here implies a new result category in the summary (`N Scenarios (X failed, Y ambiguous, ... )`